### PR TITLE
Feature - Manual Tracking

### LINKED ISSUE #76: Screen Tracking API

### PR DESCRIPTION
**Pull request by samwoodall**
Add screenViewedEventCreation function

closes #76

### CODE REVIEW

File: core/src/main/kotlin/io/rover/campaigns/core/events/domain/Event.kt [hunk 1]
```diff
@@ -14,4 +14,15 @@ data class Event(
         name: String,
         attributes: Attributes
     ): this(name, attributes, Date(), UUID.randomUUID())
+
+    companion object {
+        fun screenViewedEvent(screenName: String, screenLabel: String? = null, contentID: String? = null): Event {
+            val attributes = mutableMapOf<String, Any>("screenName" to screenName)
+
+            screenLabel?.let { attributes.put("screenLabel", it) }
+            contentID?.let { attributes.put("contentID", contentID) }
```
Comment (orospakr) [inline core/src/main/kotlin/io/rover/campaigns/core/events/domain/Event.kt]: should you use `it` here? you're grabbing the optional value out of the outer scope, by accident maybe?

Comment (samwoodall) [reply core/src/main/kotlin/io/rover/campaigns/core/events/domain/Event.kt]: Yeah, thanks should be `it`

File: core/src/main/kotlin/io/rover/campaigns/core/events/domain/Event.kt [hunk 2]
```diff
@@ -14,4 +14,15 @@ data class Event(
         name: String,
         attributes: Attributes
     ): this(name, attributes, Date(), UUID.randomUUID())
+
+    companion object {
+        fun screenViewedEvent(screenName: String, screenLabel: String? = null, contentID: String? = null): Event {
```
Comment (orospakr) [inline core/src/main/kotlin/io/rover/campaigns/core/events/domain/Event.kt]: consider naming as `screenViewed`, not `screenViewedEvent`.

File: core/src/main/kotlin/io/rover/campaigns/core/events/domain/Event.kt [hunk 3]
```diff
@@ -14,4 +14,15 @@ data class Event(
         name: String,
         attributes: Attributes
     ): this(name, attributes, Date(), UUID.randomUUID())
+
+    companion object {
+        fun screenViewed(screenName: String, contentName: String? = null, contentID: String? = null): Event {
```
Comment (orospakr) [inline core/src/main/kotlin/io/rover/campaigns/core/events/domain/Event.kt]: minor niggle, let's re-order it so contentID comes before contentName. I did the same in the iOS one

**Pull request merged**